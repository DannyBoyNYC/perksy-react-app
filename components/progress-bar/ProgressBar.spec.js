const React = require('react');
const { shallow } = require('enzyme');

const ProgressBar = require('./index').default;

describe('<ProgressBar.* />', () => {
  let component;
  const requiredProps = {
    count: 100,
    total: 1000
  };

  describe('<ProgressBar.Percentage />', () => {
    beforeEach(() => {
      component = shallow(<ProgressBar.Percentage {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName(
        'progress-bar progress-bar--percentage'
      );
    });

    it('should render a title if given one', () => {
      component = shallow(
        <ProgressBar.Percentage {...requiredProps} title="Title Text" />
      ).dive();
      expect(component.find('.progress-bar__title')).toHaveText('Title Text');
    });

    it('should render the percentage if shouldShowPercentage is true', () => {
      component = shallow(
        <ProgressBar.Percentage {...requiredProps} shouldShowPercentage />
      ).dive();
      expect(component.find('.progress-bar__percentage')).toHaveText('10%');
    });
  });
});
