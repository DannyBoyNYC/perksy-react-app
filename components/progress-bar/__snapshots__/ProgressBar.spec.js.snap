// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar.* /> <ProgressBar.Percentage /> should render correctly with all required props 1`] = `
<div
  className="progress-bar progress-bar--percentage"
>
  <div
    className="progress-bar__label"
  >
    100 / 1000
  </div>
  <div
    className="progress-bar__bar"
  >
    <div
      className="progress-bar__filler"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;
